### PR TITLE
Add more inverse transforms in rust

### PR DESCRIPTION
**Pull request by KyleSiefring**

### CODE REVIEW

**Comment by tdaede** [comment]
LGTM.

File: src/transform.rs [hunk 1]
```diff
@@ -67,39 +78,597 @@ fn av1_idct4(input: [i32; 4], output: &mut [i32], range: usize) {
...
+  output[9] = -stg8[11];
+  output[10] = stg8[15];
+  output[11] = -stg8[7];
+  output[12] = stg8[5];
+  output[13] = -stg8[13];
+  output[14] = stg8[9];
+  output[15] = -stg8[1];
+}
+
+fn av1_iidentity16(input: &[i32], output: &mut [i32], _range: usize) {
```
Comment (KyleSiefring) [inline src/transform.rs]: This was added.

File: src/transform.rs [hunk 2]
```diff
@@ -67,39 +78,597 @@ fn av1_idct4(input: [i32; 4], output: &mut [i32], range: usize) {
...
+  match tx_type {
+    TxType::DCT_DCT => Some((TxType1D::DCT, TxType1D::DCT)),
+    TxType::ADST_DCT => Some((TxType1D::ADST, TxType1D::DCT)),
+    TxType::DCT_ADST => Some((TxType1D::DCT, TxType1D::ADST)),
+    TxType::ADST_ADST => Some((TxType1D::ADST, TxType1D::ADST)),
+    TxType::IDTX => Some((TxType1D::IDTX, TxType1D::IDTX)),
+    TxType::V_DCT => Some((TxType1D::DCT, TxType1D::IDTX)),
+    TxType::H_DCT => Some((TxType1D::IDTX, TxType1D::DCT)),
+    TxType::V_ADST => Some((TxType1D::ADST, TxType1D::IDTX)),
+    TxType::H_ADST => Some((TxType1D::IDTX, TxType1D::ADST)),
```
Comment (KyleSiefring) [inline src/transform.rs]: V_ADST and H_ADST were added.

File: src/transform.rs [hunk 3]
```diff
@@ -223,19 +792,19 @@ fn iht4x4_add(
...
-        av1_inv_txfm2d_add_4x4(
-          input.as_ptr(),
-          output.as_mut_ptr(),
-          stride as libc::c_int,
-          tx_type as libc::c_int,
-          bit_depth as libc::c_int
-        );
-      }
+  if tx_type <= TxType::ADST_ADST
+    || (tx_type >= TxType::IDTX && tx_type <= TxType::H_ADST)
```
Comment (KyleSiefring) [inline src/transform.rs]: If statements no longer nested and allow more txfms

File: src/transform.rs [hunk 4]
```diff
@@ -266,7 +835,11 @@ fn iht8x8_add(
   input: &[i32], output: &mut [u16], stride: usize, tx_type: TxType, bit_depth: usize
 ) {
   // SIMD code may assert for transform types beyond TxType::IDTX.
-  if tx_type < TxType::IDTX {
+  if tx_type <= TxType::ADST_ADST
+    || (tx_type >= TxType::IDTX && tx_type <= TxType::H_ADST)
```
Comment (KyleSiefring) [inline src/transform.rs]: Here too

File: src/transform.rs [hunk 5]
```diff
@@ -307,7 +880,11 @@ fn iht16x16_add(
   input: &[i32], output: &mut [u16], stride: usize, tx_type: TxType, bit_depth: usize
 ) {
   unsafe {
-    if tx_type < TxType::IDTX {
+    if tx_type <= TxType::ADST_ADST
+      || (tx_type >= TxType::IDTX && tx_type <= TxType::H_ADST)
```
Comment (KyleSiefring) [inline src/transform.rs]: and here

**Pull request merged**